[CL-667] RuboCop on CI phase 2

koendehondt opened the pull request:
### Note to reviewers

Of course you are free to review all the changes. It is very educational. There is a lot to be learned from the changes that RuboCop has made.

But reviewing all the code will take a lot of time, because 1235 files have been changed. So it is probably better to review a selection of commits. When reviewing commits, please take into account that these commits do not include the last version of .rubocop.yml and .rubocop_todo.yml. Also know that the commits may include a .rubocop_desired.yml file that was used temporarily to keep the scope of the offences small during the process of fixing offences. Also know that some files may have changed later when layout/spacing/style offences have been corrected.

A lot of files have been corrected automatically and inspected/reviewed after the automatic correction.
A lot of changes are related to layout, spacing, and style. It is not very useful to review those.
The list below includes some commits with manual changes, because safe automatic correction was not possible.
The last commit in the list is a big one that combines a lot of changes after all cops were enabled, instead of enabling a selection of cops.
The list includes the commit message, the link to the commit, and the link to the RuboCop documentation.

Enable Style/StringConcatenation and adapt manually: https://github.com/CitizenLabDotCo/citizenlab/pull/1875/commits/9b3d90bf1fef25e4bc4ebb7213105649e1b81efe
See https://docs.rubocop.org/rubocop/cops_style.html#stylestringconcatenation

Enable Naming/VariableNumber and adapt manually: https://github.com/CitizenLabDotCo/citizenlab/pull/1875/commits/fb122a904ea0f6860a94a221a88ccf6446f38543
See https://docs.rubocop.org/rubocop/cops_naming.html#namingvariablenumber

Enable Performance/TimesMap and adapt manually: https://github.com/CitizenLabDotCo/citizenlab/pull/1875/commits/04b028f55a54de4e7a700d72edb6495ceea430e9
See https://docs.rubocop.org/rubocop-performance/cops_performance.html#performancetimesmap

Enable Style/AndOr and adapt manually: https://github.com/CitizenLabDotCo/citizenlab/pull/1875/commits/54c878dd90bfacc4095353d91f682b5a2af716c7
See https://docs.rubocop.org/rubocop/cops_style.html#styleandor

Enable all cops by default, instead of enabling selected cops: https://github.com/CitizenLabDotCo/citizenlab/pull/1875/commits/b83b4c46a7d80d5740fb46619d3c5bae8277ef15

### History
As suggested by @kogre in https://citizenlab.atlassian.net/browse/CL-667, this is the first big batch of autocorrected fixes by RuboCop. Only a few changes were made by hand when RuboCop indicated that autocorrect was unsafe.

When enabling all Cops (so not only the currently enabled ones), we came from more than 33000 offences. Now we have 10210 left.

Autocorrection takes the currently enabled Cops into account, so it is possible that some parts of the code may not have consistent indentation. That will probably be fixed later when more Cops are enabled.

After merging this PR into the master branch, and merging the master into feature branches, it is to be expected that there will be merge conflicts in the feature branches, because this PR changes a lot of files. However, when merging master into the branch several times, not many conflicts occurred, probably because the Rails code is not changed often.

## Changes

* Updated all RuboCop related gems to their latest versions.
* Ignored `back/engines/ee` otherwise the copied engines are linted too.
* Enabled all Cops except a few.
* All remaining offences are listed in the to-do file.

## How urgent is a code review?

The sooner it is reviewed, the less extra work it generates to keep it in sync with the master.

koendehondt committed:
[CL-667] Ignore back/engines/ee otherwise the copied engines are linted too

koendehondt committed:
[CL-667] Enable  Layout/SpaceInsideHashLiteralBraces

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspaceinsidehashliteralbraces

koendehondt committed:
[CL-667] Enable  Layout/SpaceInsideBlockBraces

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspaceinsideblockbraces

koendehondt committed:
[CL-667] Enable Layout/SpaceAroundOperators

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspacearoundoperators

koendehondt committed:
[CL-667] Enable Style/MethodDefParentheses

https://docs.rubocop.org/rubocop/cops_style.html#stylemethoddefparentheses

koendehondt committed:
[CL-667] Enable Layout/SpaceBeforeBlockBraces

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspacebeforeblockbraces

koendehondt committed:
[CL-667] Enable Layout/SpaceAfterComma

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspaceaftercomma

koendehondt committed:
[CL-667] Enable Layout/TrailingWhitespace

https://docs.rubocop.org/rubocop/cops_layout.html#layouttrailingwhitespace

koendehondt committed:
[CL-667] Enable Layout/IndentationStyle

https://docs.rubocop.org/rubocop/cops_layout.html#layoutindentationstyle

koendehondt committed:
[CL-667] Enable Layout/EmptyLinesAroundBlockBody

https://docs.rubocop.org/rubocop/cops_layout.html#layoutemptylinesaroundblockbody

koendehondt committed:
[CL-667] Enable Style/BlockDelimiters

https://docs.rubocop.org/rubocop/cops_style.html#styleblockdelimiters

koendehondt committed:
[CL-667] Enable Layout/EmptyLinesAroundClassBody

https://docs.rubocop.org/rubocop/cops_layout.html#layoutemptylinesaroundclassbody

koendehondt committed:
[CL-667] Disable Layout/HashAlignment because it requires discussion

https://docs.rubocop.org/rubocop/cops_layout.html#layouthashalignment

koendehondt committed:
[CL-667] Disable  Layout/FirstHashElementIndentation because the setting is dependent on Layout/HashAlignment

https://docs.rubocop.org/rubocop/cops_layout.html#layoutfirsthashelementindentation

koendehondt committed:
[CL-667] Enable Layout/EmptyLines

https://docs.rubocop.org/rubocop/cops_layout.html#layoutemptylines

koendehondt committed:
[CL-667] Enable Layout/TrailingEmptyLines

https://docs.rubocop.org/rubocop/cops_layout.html#layouttrailingemptylines

koendehondt committed:
[CL-667] Enable Layout/IndentationConsistency

https://docs.rubocop.org/rubocop/cops_layout.html#layoutindentationconsistency

koendehondt committed:
[CL-667] Layout/MultilineBlockLayout

https://docs.rubocop.org/rubocop/cops_layout.html#layoutmultilineblocklayout

koendehondt committed:
[CL-667] Enable Layout/ExtraSpacing

https://docs.rubocop.org/rubocop/cops_layout.html#layoutextraspacing

koendehondt committed:
[CL-667] Enable Layout/EmptyLineAfterGuardClause

https://docs.rubocop.org/rubocop/cops_layout.html#layoutemptylineafterguardclause

koendehondt committed:
[CL-667] Enable Style/PercentLiteralDelimiters

https://docs.rubocop.org/rubocop/cops_style.html#stylepercentliteraldelimiters

koendehondt committed:
[CL-667] Layout/SpaceAfterColon

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspaceaftercolon

koendehondt committed:
[CL-667] Layout/CaseIndentation

https://docs.rubocop.org/rubocop/cops_layout.html#layoutcaseindentation

koendehondt committed:
[CL-667] Enable Style/Semicolon

https://docs.rubocop.org/rubocop/cops_style.html#stylesemicolon

koendehondt committed:
[CL-667] Enable Layout/LeadingEmptyLines

https://docs.rubocop.org/rubocop/cops_layout.html#layoutleadingemptylines

koendehondt committed:
[CL-667] Enable Layout/AssignmentIndentation

https://docs.rubocop.org/rubocop/cops_layout.html#layoutassignmentindentation

koendehondt committed:
[CL-667] Layout/SpaceAroundBlockParameters

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspacearoundblockparameters

koendehondt committed:
[CL-667] Enable Layout/SpaceBeforeComment

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspacebeforecomment

koendehondt committed:
[CL-667] Enable Layout/EmptyLinesAroundExceptionHandlingKeywords

https://docs.rubocop.org/rubocop/cops_layout.html#layoutemptylinesaroundclassbody

koendehondt committed:
[CL-667] Enable Style/RedundantSelf

https://docs.rubocop.org/rubocop/cops_style.html#styleredundantself

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

# Conflicts:
#	back/engines/commercial/project_folders/app/services/project_folders/side_fx_service.rb
#	back/engines/commercial/smart_groups/spec/services/side_fx_project_spec.rb

koendehondt committed:
[CL-667] Enable Style/TrailingCommaInHashLiteral

https://docs.rubocop.org/rubocop/cops_style.html#styletrailingcommainhashliteral

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] bundle update rubocop rubocop-ast rubocop-performance rubocop-rails rubocop-rspec

koendehondt committed:
[CL-667] AllowInHeredoc in Layout/TrailingWhitespace

koendehondt committed:
[CL-667] Disable Style/Documentation as most classes are not documented

https://docs.rubocop.org/rubocop/cops_style.html#styledocumentation

koendehondt committed:
[CL-667] Add Metrics/MethodLength to .rubocop_desired.yml, and .rubocop_todo.yml, as it requires manual changes to fix offences

https://docs.rubocop.org/rubocop/cops_metrics.html#metricsmethodlength

koendehondt committed:
[CL-667] Remove app/models/.rubocop.yml

koendehondt committed:
[CL-667] Enable Metrics/ClassLength

https://docs.rubocop.org/rubocop/cops_metrics.html#metricsclasslength

koendehondt committed:
[CL-667] Add desired Style/FrozenStringLiteralComment

https://docs.rubocop.org/rubocop/cops_style.html#stylefrozenstringliteralcomment

koendehondt committed:
[CL-667] Add desired Layout/LineLength

https://docs.rubocop.org/rubocop/cops_layout.html#layoutlinelength

koendehondt committed:
[CL-667] Add desired Metrics/BlockLength

https://docs.rubocop.org/rubocop/cops_metrics.html#metricsblocklength

koendehondt committed:
[CL-667] Disable RSpec/MultipleExpectations

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecmultipleexpectations

koendehondt committed:
[CL-667] Add desired RSpec/ExampleLength, RSpec/InstanceVariable, RSpec/EmptyExampleGroup, RSpec/DescribedClass, RSpec/ContextWording, RSpec/MultipleMemoizedHelpers

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecexamplelength
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecemptyexamplegroup
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecinstancevariable
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecdescribedclass
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspeccontextwording
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecmultiplememoizedhelpers

koendehondt committed:
[CL-667] Enable Style/WordArray

https://docs.rubocop.org/rubocop/cops_style.html#stylewordarray

koendehondt committed:
[CL-667] Enable RSpec/ImplicitExpect

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecimplicitexpect

koendehondt committed:
[CL-667] Add desired Style/ClassAndModuleChildren

https://docs.rubocop.org/rubocop/cops_style.html#styleclassandmodulechildren

koendehondt committed:
[CL-667] Add desired RSpec/MissingExampleGroupArgument, RSpec/LetSetup, RSpec/MessageSpies, RSpec/FilePath, RSpec/ExampleWording

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecmissingexamplegroupargument
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecletsetup
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecmessagespies
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecfilepath
https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecfilepath

koendehondt committed:
[CL-667] Enable RSpec/EmptyLineAfterHook

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecemptylineafterhook

koendehondt committed:
[CL-667] Enable RSpec/ScatteredLet

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecscatteredlet

koendehondt committed:
[CL-667] Enable RSpec/EmptyLineAfterFinalLet

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecemptylineafterfinallet

koendehondt committed:
[CL-667] Add desired RSpec/PredicateMatcher

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecpredicatematcher

koendehondt committed:
[CL-667] Add desired RSpec/ScatteredSetup

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecscatteredsetup

koendehondt committed:
[CL-667] Add desired RSpec/EmptyLineAfterFinalLet, RSpec/NamedSubject, RSpec/AnyInstance, RSpec/NestedGroups,  RSpec/RepeatedDescription, RSpec/DescribeClass

koendehondt committed:
[CL-667] Enable RSpec/EmptyLineAfterExample

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecemptylineafterexample

koendehondt committed:
[CL-667] Add spec type excluded by RSpec/DescribeClass

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Add desired Lint/UselessAssignment, Style/GuardClause, Metrics/AbcSize, Style/MutableConstant

https://docs.rubocop.org/rubocop/cops_lint.html#lintuselessassignment
https://docs.rubocop.org/rubocop/cops_style.html#styleguardclause
https://docs.rubocop.org/rubocop/cops_metrics.html#metricsabcsize
https://docs.rubocop.org/rubocop/cops_style.html#stylemutableconstant

sebastienhoorens commented on back/.rubocop.yml:
```diff
@@ -10,10 +14,134 @@ AllCops:
+  Enabled: true
+
+Layout/HashAlignment:
+  # There are a lot of places where EnforcedColonStyle: table is expected.
```
I played a bit with this rule once, and I think it was possible to support multiple configurations (for example `[key, table]`).

koendehondt replied:
Yes, it is possible. I did not apply it yet because of the impact on layout.

sebastienhoorens commented on back/.rubocop_todo.yml:
```diff
@@ -0,0 +1,202 @@
+# Configuration parameters: IgnoredMetadata.
+RSpec/DescribeClass:
+  Exclude:
+    - 'engines/commercial/admin_api/spec/graphql/folders_spec.rb'
```
Do these paths work both on the local machine (where the file paths start from "back" - e.g. in the code editor) as well as inside the docker containers (e.g. when running in CI)?

koendehondt replied:
I ran all rubocop commands locally in a docker container. But VSCode does not. So this is a good point. I will look into it.

koendehondt replied:
It works in VSCode, because the paths are relative to the folder in which rubocop.yml resides.

sebastienhoorens commented on back/app/controllers/web_api/v1/groups_projects_controller.rb:
```diff
@@ -7,14 +7,14 @@ def index
-        @groups_projects
-      else
-        raise 'Unsupported sort method'
+    when 'new'
```
I like the indentation in this case, but I can definitely live with this change.

sebastienhoorens reviewed (approved):
It's really nice to see so much code could be fixed automatically!

koendehondt committed:
[CL-667] Enable RSpec/NotToNot

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecnottonot

koendehondt committed:
[CL-667] Add desired Style/IfUnlessModifier

https://docs.rubocop.org/rubocop/cops_style.html#styleifunlessmodifier

koendehondt committed:
[CL-667] Use'request' type in RSpec/DescribeClass IgnoredMetadata, because 'verification' makes a spec fail

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Add desired Layout/MultilineMethodCallIndentation

https://docs.rubocop.org/rubocop/cops_layout.html#layoutmultilinemethodcallindentation

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

# Conflicts:
#	back/app/services/text_image_service.rb
#	back/spec/services/text_image_service_spec.rb

koendehondt committed:
[CL-667] Enable Style/HashSyntax

https://docs.rubocop.org/rubocop/cops_style.html#stylehashsyntax

koendehondt committed:
[CL-667] [CL-667] Lint/UnusedMethodArgument

https://docs.rubocop.org/rubocop/cops_lint.html#lintunusedmethodargument

koendehondt committed:
[CL-667] Enable Lint/ParenthesesAsGroupedExpression

https://docs.rubocop.org/rubocop/cops_lint.html#lintparenthesesasgroupedexpression

koendehondt committed:
[CL-667] Enable Style/SymbolArray

https://docs.rubocop.org/rubocop/cops_style.html#stylesymbolarray

koendehondt committed:
[CL-667] Enable Layout/ClosingParenthesisIndentation

https://docs.rubocop.org/rubocop/cops_layout.html#layoutclosingparenthesisindentation

koendehondt committed:
[CL-667] Enable Style/SingleArgumentDig

https://docs.rubocop.org/rubocop/cops_style.html#stylesingleargumentdig

koendehondt committed:
[CL-667] Add desired Rails/TimeZone

https://www.rubydoc.info/gems/rubocop/0.41.2/RuboCop/Cop/Rails/TimeZone

koendehondt committed:
[CL-667] Add desired Layout/BlockEndNewline

https://docs.rubocop.org/rubocop/cops_layout.html#layoutblockendnewline

koendehondt committed:
[CL-667] Add desired Layout/ArgumentAlignment

https://docs.rubocop.org/rubocop/cops_layout.html#layoutargumentalignment

koendehondt committed:
[CL-667] Add desired Layout/FirstArgumentIndentation, Layout/FirstArrayElementIndentation, Layout/ArrayAlignment

https://docs.rubocop.org/rubocop/cops_layout.html#layoutfirstargumentindentation
https://docs.rubocop.org/rubocop/cops_layout.html#layoutfirstarrayelementindentation
https://docs.rubocop.org/rubocop/cops_layout.html#layoutarrayalignment

koendehondt committed:
[CL-667] Enable Style/NegatedIf

https://docs.rubocop.org/rubocop/cops_style.html#stylenegatedif

koendehondt committed:
[CL-667] Enable Layout/SpaceAroundEqualsInParameterDefault

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspacearoundequalsinparameterdefault

koendehondt committed:
[CL-667] Add desired Metrics/CyclomaticComplexity, Layout/MultilineMethodCallBraceLayout

https://docs.rubocop.org/rubocop/cops_layout.html#layoutmultilinemethodcallbracelayout
https://docs.rubocop.org/rubocop/cops_metrics.html#metricscyclomaticcomplexity

koendehondt committed:
[CL-667] Enable Style/StringLiteralsInInterpolation

https://docs.rubocop.org/rubocop/cops_style.html#stylestringliteralsininterpolation

koendehondt committed:
[CL-667] Add desired Lint/SafeNavigationConsistency

https://docs.rubocop.org/rubocop/cops_lint.html#lintsafenavigationconsistency

koendehondt committed:
[CL-667] Add desired Naming/MethodParameterName

https://docs.rubocop.org/rubocop/cops_naming.html#namingmethodparametername

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Enable Layout/EmptyLinesAroundModuleBody

https://docs.rubocop.org/rubocop/cops_layout.html#layoutemptylinesaroundmodulebody

koendehondt committed:
[CL-667] Enable Lint/UnusedBlockArgument

https://docs.rubocop.org/rubocop/cops_lint.html#lintunusedblockargument

koendehondt committed:
[CL-667] Add desired Metrics/PerceivedComplexity

https://docs.rubocop.org/rubocop/cops_metrics.html#metricsperceivedcomplexity

koendehondt committed:
[CL-667] Enable Rails/HttpStatus

https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/Rails/HttpStatus

koendehondt committed:
[CL-667] Add desired Layout/IndentationWidth

https://docs.rubocop.org/rubocop/cops_layout.html#layoutindentationwidth

koendehondt committed:
[CL-667] Enable Style/Proc

https://docs.rubocop.org/rubocop/cops_style.html#styleproc

koendehondt committed:
[CL-667] Add desired Gemspec/RequiredRubyVersion

https://docs.rubocop.org/rubocop/cops_gemspec.html#gemspecrequiredrubyversion

koendehondt committed:
[CL-667] Enable Style/TrailingCommaInArrayLiteral

https://docs.rubocop.org/rubocop/cops_style.html#styletrailingcommainarrayliteral

koendehondt committed:
[CL-667] Add desired Rails/SkipsModelValidations

https://www.rubydoc.info/gems/rubocop/0.51.0/RuboCop/Cop/Rails/SkipsModelValidations

koendehondt committed:
[CL-667] Add desired Gemspec/OrderedDependencies, Rails/ReversibleMigration

https://docs.rubocop.org/rubocop/cops_gemspec.html#gemspecordereddependencies
https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsreversiblemigration

koendehondt committed:
[CL-667] Enable Style/SpecialGlobalVars

https://docs.rubocop.org/rubocop/cops_style.html#stylespecialglobalvars

koendehondt committed:
[CL-667] Add desired Rails/InverseOf

https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsinverseof

koendehondt committed:
[CL-667] Enable Rails/PluralizationGrammar

https://docs.rubocop.org/rubocop-rails/cops_rails.html#railspluralizationgrammar

koendehondt committed:
[CL-667] Add desired Layout/EndAlignment

https://docs.rubocop.org/rubocop/cops_layout.html#layoutendalignment

koendehondt committed:
[CL-667] Added desired RSpec/StubbedMock

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecstubbedmock

koendehondt committed:
[CL-667] Enable Style/EmptyMethod

https://docs.rubocop.org/rubocop/cops_style.html#styleemptymethod

koendehondt committed:
[CL-667] Add desired Lint/AmbiguousBlockAssociation

https://docs.rubocop.org/rubocop/cops_lint.html#lintambiguousblockassociation

koendehondt committed:
[CL-667] Enable Layout/DotPosition

https://docs.rubocop.org/rubocop/cops_layout.html#layoutdotposition

koendehondt committed:
[CL-667] Enable Performance/Detect

https://docs.rubocop.org/rubocop-performance/cops_performance.html#performancedetect

koendehondt committed:
[CL-667] Enable Style/ExpandPathArguments

https://docs.rubocop.org/rubocop/cops_style.html#styleexpandpatharguments

koendehondt committed:
[CL-667] enable Style/NestedParenthesizedCalls

https://docs.rubocop.org/rubocop/cops_style.html#stylenestedparenthesizedcalls

koendehondt committed:
[CL-667] Enable Layout/EmptyLinesAroundMethodBody

https://docs.rubocop.org/rubocop/cops_layout.html#layoutemptylinesaroundmethodbody

koendehondt committed:
[CL-667] Add desired Layout/MultilineHashBraceLayout

https://docs.rubocop.org/rubocop/cops_layout.html#layoutmultilinehashbracelayout

koendehondt committed:
[CL-667] Enable Style/CommentAnnotation

https://docs.rubocop.org/rubocop/cops_style.html#stylecommentannotation

koendehondt committed:
[CL-667] Add desired Layout/ElseAlignment

https://docs.rubocop.org/rubocop/cops_layout.html#layoutelsealignment

koendehondt committed:
[CL-667] Enable Naming/VariableNumber and adapt manually

https://docs.rubocop.org/rubocop/cops_naming.html#namingvariablenumber

koendehondt committed:
[CL-667] Enable Style/SymbolProc, with some exclusions where it is inappropriate

https://docs.rubocop.org/rubocop/cops_style.html#stylesymbolproc

koendehondt committed:
[CL-667] Enable Layout/SpaceInLambdaLiteral

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspaceinlambdaliteral

koendehondt committed:
[CL-667] Disable Style/NumericPredicate

https://docs.rubocop.org/rubocop/cops_style.html#stylenumericpredicate

koendehondt committed:
[CL-667] Enable Style/NumericLiterals

https://docs.rubocop.org/rubocop/cops_style.html#stylenumericliterals

koendehondt committed:
[CL-667] Enable Style/TrailingCommaInArguments

https://docs.rubocop.org/rubocop/cops_style.html#styletrailingcommainarguments

koendehondt committed:
[CL-667] Enable Layout/SpaceInsideParens

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspaceinsideparens

koendehondt committed:
[CL-667] Enable Layout/SpaceInsideArrayLiteralBrackets

https://docs.rubocop.org/rubocop/cops_layout.html#layoutspaceinsidearrayliteralbrackets

koendehondt committed:
[CL-667] Enable Style/ClassCheck

https://docs.rubocop.org/rubocop/cops_style.html#styleclasscheck

koendehondt committed:
[CL-667] Enable Performance/TimesMap and adapt manually

https://docs.rubocop.org/rubocop-performance/cops_performance.html#performancetimesmap

koendehondt committed:
[CL-667] Enable Rails/IndexWith

https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsindexwith

koendehondt committed:
[CL-667] Add desired Naming/PredicateName

https://docs.rubocop.org/rubocop/cops_naming.html#namingpredicatename

koendehondt committed:
[CL-667] Enable RSpec/EmptyLineAfterSubject

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecemptylineaftersubject

koendehondt committed:
[CL-667] Enable Style/Lambda

https://docs.rubocop.org/rubocop/cops_style.html#stylelambda

koendehondt committed:
[CL-667] Enable Style/AndOr and adapt manually

https://docs.rubocop.org/rubocop/cops_style.html#styleandor

koendehondt committed:
[CL-667] Enable Style/EmptyElse

https://docs.rubocop.org/rubocop/cops_style.html#styleemptyelse

koendehondt committed:
[CL-667] Enable Style/OneLineConditional

https://docs.rubocop.org/rubocop/cops_style.html#styleonelineconditional

koendehondt committed:
[CL-667] Enable Style/StringConcatenation and adapt manually

https://docs.rubocop.org/rubocop/cops_style.html#stylestringconcatenation

koendehondt committed:
[CL-667] Add desired Rails/Date, Layout/BlockAlignment, Layout/BeginEndAlignment, Rails/CreateTableWithTimestamps, Style/MultilineBlockChain, Rails/HasManyOrHasOneDependent

https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsdate
https://docs.rubocop.org/rubocop/cops_layout.html#layoutblockalignment
https://docs.rubocop.org/rubocop/cops_layout.html#layoutbeginendalignment
https://docs.rubocop.org/rubocop-rails/cops_rails.html#railscreatetablewithtimestamps
https://docs.rubocop.org/rubocop-rails/cops_rails.html#railshasmanyorhasonedependent

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Enable RSpec/EmptyLineAfterExampleGroup

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecemptylineafterexamplegroup

koendehondt committed:
[CL-667] Enable Style/RegexpLiteral

https://docs.rubocop.org/rubocop/cops_style.html#styleregexpliteral

koendehondt committed:
[CL-667] Enable Style/RedundantFileExtensionInRequire

https://docs.rubocop.org/rubocop/cops_style.html#styleredundantfileextensioninrequire

koendehondt committed:
[CL-667] Enable Layout/EmptyLineAfterMagicComment

https://docs.rubocop.org/rubocop/cops_layout.html#layoutemptylineaftermagiccomment

koendehondt committed:
[CL-667] Add desired RSpec/RepeatedExampleGroupDescription, Style/ClassVars

https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecrepeatedexamplegroupdescription
https://docs.rubocop.org/rubocop/cops_style.html#styleclassvars

koendehondt committed:
[CL-667] Enable Style/RedundantRegexpEscape

https://docs.rubocop.org/rubocop/cops_style.html#styleredundantregexpescape

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Enable Style/Next

https://docs.rubocop.org/rubocop/cops_style.html#stylenext

koendehondt committed:
[CL-667] Enable all remaining Style Cops, except Style/DoubleNegation

koendehondt committed:
[CL-667] Enable all remaining Performance Cops

koendehondt committed:
[CL-667] Add desired remaining Metrics Cops

koendehondt committed:
[CL-667] Enable all remaining Naming Cops, but put Naming/AccessorMethodName in desired list

koendehondt committed:
[CL-667] Fix wrong autocorrect when enabling remaining Style Cops

Mistake reported by tests on CI.

koendehondt committed:
[CL-667] Enable all remaining Layout Cops

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] [CL-667] Enable all remaining Lint Cops, except Lint/MissingSuper

koendehondt committed:
[CL-667] Enable all remaining Rails Cops, and add desired Rails/CreateTableWithTimestamps, Rails/OutputSafety, Rails/LexicallyScopedActionFilter, Rails/HasManyOrHasOneDependent

koendehondt committed:
[CL-667] Fix silly typo and wrong autocorrections in previous commit

koendehondt committed:
[CL-667] Add desired Rails/HasManyOrHasOneDependent

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

# Conflicts:
#	back/app/services/sanitization_service.rb

koendehondt committed:
[CL-667] Enable all remaining RSpec Cops and add desired RSpec/BeforeAfterAll, RSpec/ExpectInHook, RSpec/RepeatedExample, RSpec/VerifiedDoubles

koendehondt committed:
[CL-667] Add desired Bundler/OrderedGems, Security/JSONLoad, Security/Open, Security/YAMLLoad

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

# Conflicts:
#	back/Gemfile.lock

koendehondt committed:
[CL-667] After merging master, bundle update rubocop rubocop-ast rubocop-performance rubocop-rails rubocop-rspec

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Enable all cops by default, instead of enabling selected cops

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Fix Performance/MapCompact

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

nTraum commented on back/.rubocop.yml:
```diff
@@ -1,96 +1,103 @@
+
+Layout/HashAlignment:
+  # There are a lot of places where EnforcedColonStyle: table is expected.
+  # Discussion needed, because the default will affect a lot of table like hash literals.
```
My 2 cents: I'd prefer to stay as close to the default Rubocop config as possible where we can and spare us these endless discussions and let an auto-formatter do that job for me instead.

koendehondt replied:
I agree, but I was not sure that everybody would 😄 I will take out the comment. Note that I cannot take out `Enabled: false` now, because that would result in many code formatting changes. We will keep it for a next PR.

nTraum commented on back/.rubocop.yml:
```diff
@@ -1,96 +1,103 @@
+    type: ['request']
+
+Metrics/MethodLength:
+  # Max: 30
```
Shall we remove that comment @koendehondt?

koendehondt replied:
I kept it, because we need a discussion on it. Currently anything goes.
But I agree that it is unclear why it is commented out. I will take it out.

koendehondt committed:
[CL-667] Apply code review suggestions

nTraum reviewed (approved):
Let's go! :)

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

# Conflicts:
#	back/app/services/participation_context_service.rb

koendehondt committed:
[CL-667] Make RuboCop happy

sebastienhoorens commented:
Still all good for me :) Awesome work!

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

koendehondt committed:
[CL-667] Update changelog

koendehondt committed:
[CL-667] Merge branch 'master' into CL-667-rubocop-on-ci-phase-2

# Conflicts:
#	CHANGELOG.md
